### PR TITLE
feat(workflow): Create alert from event.type error/default

### PR DESCRIPTION
**Pull request by scttcper**
Handle creating alerts with `event.type:default OR event.type:error` and removing that piece from the query when creating from alerts. Should fill in the event source dropdown.

Should not allow `event.type:transaction or event.type:error`.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/alerts/utils/index.tsx [hunk 1]
```diff
@@ -209,3 +209,35 @@ export function convertDatasetEventTypesToSource(
     return Datasource.ERROR;
   }
 }
+
+/**
+ * Attempt to guess the data source of a discover query
+ *
+ * @returns A new query without the datasource. null on no datasource
```
Comment (davidenwang) [inline src/sentry/static/sentry/app/views/alerts/utils/index.tsx:216]: nit: change comment to something like `a new object containing the datasource and query without datasource...`

File: tests/js/spec/views/alerts/utils.spec.jsx [hunk 2]
```diff
@@ -81,4 +82,69 @@ describe('Alert utils', function () {
...
+        )
+      ).toEqual({
+        source: Datasource.ERROR_DEFAULT,
+        query: 'transaction.duration:<30s',
+      });
+      expect(
+        getQueryDatasource('event.type:error OR (event.type:default event.level:fatal)')
+      ).toEqual({
+        source: Datasource.ERROR_DEFAULT,
+        query: 'event.level:fatal)',
```
Comment (davidenwang) [inline tests/js/spec/views/alerts/utils.spec.jsx:104]: I guess this query is pretty hard to parse, but is the unmatched parentheses expected. should we do any additional validation/cleaning steps after parsing out the datasource?

**Review by davidenwang — APPROVED** [review]
did some manual tests and looks pretty good. Not every discover query with event types can be translated into an equivalent alert so there's some weird edge cases, but idk how much it's worth putting the time into to fix all the scenarios.

File: tests/js/spec/views/alerts/utils.spec.jsx (hunk 2 above)
Comment (scttcper) [reply tests/js/spec/views/alerts/utils.spec.jsx:104]: yeah expected for this version
